hw/top_earlgrey/ip/ast/doc/_index.md

```diff
@@ -25,772 +22,875 @@ shown in the diagram, but will be explained as well.
+</tr>
+<tr class="header">
+<th>adc_pd_i</th>
+<th>I</th>
+<th>1</th>
+<th>adc</th>
+<th>ADC power down - for saving power during deep-sleep state between
+measurements. When this signal is high, ADC module is in off state,
+otherwise, it is in active state. For further description about adc_pd_i
+usage, see ADC module description below.</th>
 </tr>
 <tr class="odd">
-<td>rng_val_o</td>
-<td>O</td>
-<td>1</td>
-<td>rng</td>
-<td>RNG bit valid. This is a per-transaction valid. rng_b_o can be sampled whenever this bit is high.</td>
-</tr>
-<tr class="even">
-<td>rng_b_o</td>
-<td>O</td>
-<td>4</td>
-<td>rng</td>
-<td>RNG digital bit streams. The downstream controller of this signal should sample at the rates of 200 KHz / 100KHz / 50KHz / 25KHz / 12.5KHz (50 KHz is the target, the rest are for fallback and upside).</td>
+<th>entropy_req_o</th>
+<th>O</th>
+<th>edn_req</th>
+<th>es</th>
+<th>Request entropy from EDN</th>
+</tr>
+<tr class="header">
+<th>entropy_rsp_i</th>
+<th>I</th>
+<th>edn_rsp</th>
+<th>es</th>
+<th>EDN entropy request acknowledgement and data.</th>
 </tr>
 <tr class="odd">
-<td><strong>Countermeasures and Alerts</strong></td>
-<td></td>
-<td></td>
-<td></td>
-<td></td>
-</tr>
-<tr class="even">
-<td>as_alert_po/no</td>
-<td>O</td>
-<td>2</td>
-<td>alert</td>
-<td>Active shield alert</td>
+<th>rng_en_i</th>
+<th>I</th>
+<th>1</th>
+<th>rng</th>
+<th>Input from controller to enable RNG</th>
+</tr>
+<tr class="header">
+<th>rng_fips_i</th>
+<th>I</th>
+<th>1</th>
+<th>rng</th>
+<th>Indicates that the AST RNG module is requested to output FIPS
+SP-800-90B grade RNG bits. This may, but not necessarily affect
+bit-rate. This bit is a placeholder. The use of this signal inside AST
+is TBD.</th>
 </tr>
 <tr class="odd">
-<td>as_alert_ack_i</td>
-<td>I</td>
-<td>1</td>
-<td>alert</td>
-<td>single pulse ack, source and destination assumed to be synchronous</td>
-</tr>
-<tr class="even">
-<td>as_alert_trig_i</td>
-<td>I</td>
-<td>1</td>
-<td>alert</td>
-<td>Alert force trigger by software - connected to a chip register</td>
+<th>rng_val_o</th>
+<th>O</th>
+<th>1</th>
+<th>rng</th>
+<th>RNG bit valid. This is a per-transaction valid. rng_b_o can be
+sampled whenever this bit is high.</th>
+</tr>
+<tr class="header">
+<th>rng_b_o</th>
+<th>O</th>
+<th>4</th>
+<th>rng</th>
+<th>RNG digital bit streams. The downstream controller of this signal
+should sample the rng_b_o whenever rng_val_o is high.</th>
 </tr>
 <tr class="odd">
-<td>cg_alert_po/no</td>
-<td>O</td>
-<td>2</td>
-<td>alert</td>
-<td>Clock glitch detector alert</td>
-</tr>
-<tr class="even">
-<td>cg_alert_ack_i</td>
-<td>I</td>
-<td>1</td>
-<td>alert</td>
-<td>single pulse ack, source and destination assumed to be synchronous</td>
+<th colspan="5"><strong>Countermeasures and Alerts</strong></th>
+</tr>
+<tr class="header">
+<th>alert_req_o</th>
+<th>O</th>
+<th>ast_alert_req</th>
+<th>alert</th>
+<th>Alert events. There are 13 such events. The alerts are associated

```

Comment:
Is 13 still correct? I thought we reduced the number of alert channels to 11 after removing the flash and otp connections.